azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -595,6 +601,11 @@ private void finalizeFlow(final ExecutableFlowBase flow) {
             + durationSec + " seconds");
         flow.setStatus(Status.FAILED);
         break;
+      case KILLING:
+        this.logger
+            .info("Setting flow '" + id + "' status to KILLED in " + durationSec + " seconds");
+        flow.setStatus(Status.KILLED);
```

Comment:
so we rely on finalizeFlow to set to killed status from killing. @jamiesjc will anything cause finalizeFlow not being called for a flow?